src/Components/DatePicker/Component.purs

```diff
@@ -0,0 +1,428 @@
+                -- queries in the parent. Let's do that here. To make this work, remember
+                -- you're writing in `Select`'s HTML type and you have to wrap your constructors
+                -- in Raise.
+                arrowButton q =
+                  Button.buttonClear
+                    [ HE.onClick $ Select.always $ Select.raise $ H.action q
+                    , css "text-grey-70 p-3"
+                    ]
+
+                -- Show the month and year
+                dateHeader =
+                  HH.div
+                    [ css "flex-1" ]
+                    [ HH.text monthYear ]
+
+            calendarHeader =
+              HH.div
+                [ css "flex text-grey-70" ]
+                ( headers <#>
+                  \day ->
+                    HH.div
+                      [ css "w-14 h-14 flex items-center justify-center" ]
+                      [ HH.text day ]
+                )
+              where
+                headers = [ "S", "M", "T", "W", "T", "F", "S" ]
+
+            -- Here we'll render out our dates as rows in the calendar.
+            renderRows =
+              mapWithIndex (\row subArr -> renderRow (row * 7) subArr)
+              where
+                renderRow offset items =
+                  HH.div
+                    [ css "flex font-light" ]
+                    ( mapWithIndex
+                      (\column item -> renderItem (column + offset) item) items
+                    )
+
+            renderItem index item =
+              HH.div
+              -- Here's the place to use info from the item to render it in different
+              -- states.
+              -- if highlightedIndex == Just index then 'highlight' else 'dont'
+              -- Because there are so many possible states, what about a helper like
+              -- getCalendarStyles?
+                ( maybeSetItemProps index item
+                  [ css
+                    $ trim
+                    $ "w-14 h-14 rounded-full relative "
+                      <> "flex items-center justify-center "
+                      <> "transition-1/4 border border-white "
+                      <> "before:no-content before:transition-1/4 "
+                      <> "before:w-full before:h-full "
+                      <> "before:absolute before:pin-t before:pin-l "
+                      <> (getCalendarStyles item)
+                  ]
+                )
+                -- printDay will format our item correctly
+                [ HH.text $ printDay item ]
+              where
+                -- If the calendar item is selectable,
+                -- then augment the props with the correct click events.
+                -- if not, then just don't provide the props at all.
+                -- this is an easy way to "disable" functionality in the calendar.
+                maybeSetItemProps i (CalendarItem Selectable _ _ _) props =
```

Comment:
Do we actually want this, or do we only want to disable say the click event? Do we want to disable all props for the `disabled` items in the calendar?